imputation/src/mean/mean_test.go

```diff
@@ -0,0 +1,163 @@
+package main
+
+import (
+    "testing"
+)
+
+func TestMean(t *testing.T) {
+    // test 1 empty series
+    series1 := make(map [string]float64)
+    want := series1
+    got := fillMean(series1)
+    if  len(got) != len(want) {
+        t.Errorf("Size mismatch: fillMean() = %v, want %v", got, want)
+    }
+    
+    for k,v := range got{
+        if vw, ok := want[k]; !ok || vw != v {
+            t.Errorf("Value mismatch: fillMean() = %v, want %v", got, want)
+        }
+    } 
+    
+    //test 2 small series
+    series2 := make(map [string]float64)
+    series2["2003"] = 42.5
+    want = series2
+    got = fillMean(series2)
+    if  len(got) != len(want) {
+        t.Errorf("Size mismatch: fillMean() = %v, want %v", got, want)
+    }
+    
+    for k,v := range got{
+        if vw, ok := want[k]; !ok || vw != v {
+            t.Errorf("Value mismatch: fillMean() = %v, want %v", got, want)
+        }
+    } 
+    
+    
+    //test 3 year gap
+    series3 := make(map [string]float64)
+    series3["2003"] = 42.5
+    series3["2002"] = 18.9
+    series3["1998"] = 28
+    
+    want = make(map [string]float64)
+    want["2003"] = 42.5
+    want["2002"] = 18.9
+    want["1998"] = 28
+   
+    
+    var mean float64 = 0
+    for _,v := range series3{
+        mean += v
+    }
+    mean = mean/float64(len(series3))
+    
+    want["2000"] = mean
+    want["2001"] = mean
+    want["1999"] = mean
+    
+    got = fillMean(series3)
+    if  len(got) != len(want) {
+        t.Errorf("Size mismatch: fillMean() = %v, want %v", got, want)
+    }
+    
+    for k,v := range got{
+        if vw, ok := want[k]; !ok || vw != v {
+            t.Errorf("Value mismatch: fillMean() = %v, want %v", got, want)
+        }
+    } 
+    
+    
+    //test 4 month gap
+    series4 := make(map [string]float64)
```

Comment:
The pattern seems to be:
Create an input time series.
Create an output time series by adding some values to the input time series. 

I think we can simplify the code like this:
```
series4: TimeSeries{
  "2003-02": 42.5,
  "2002-02": 18.9,
  "2003-04": 28,
}
want = createOutput(series4, ["2002-04", "2002-06", "2002-08", "2002-10", "2002-12"])
```

```
func createOutput(ts TimeSeries, desiredValues) TimeSeries {
  out := make(TimeSeries, len(ts) + len(desiredValues))
  for k, v := range ts {
    out[k] = v
  }
  m := mean(ts)
  for _, k := range(desiredValues) {
     out[k] = m
  }
  return(out)
}
```

I think this works for all five test instances.

Reply:
Thank you so much!!! This will help readability and future testing a lot.